samples/snippets/sample_preemptible.py

```diff
@@ -0,0 +1,148 @@
+    instance.name = instance_name
+    instance.disks = [disk]
+    instance.machine_type = f"zones/{zone}/machineTypes/e2-small"
+    instance.network_interfaces = [network_interface]
+
+    # Set the preemptible setting
+    instance.scheduling = compute_v1.Scheduling()
+    instance.scheduling.preemptible = True
+
+    # Prepare the request to insert an instance.
+    request = compute_v1.InsertInstanceRequest()
+    request.zone = zone
+    request.project = project_id
+    request.instance_resource = instance
+
+    # Wait for the create operation to complete.
+    print(f"Creating the {instance_name} instance in {zone}...")
+    operation = instance_client.insert_unary(request=request)
+    while operation.status != compute_v1.Operation.Status.DONE:
+        operation = operation_client.wait(
+            operation=operation.name, zone=zone, project=project_id
+        )
+    if operation.error:
+        print("Error during creation:", operation.error, file=sys.stderr)
+    if operation.warnings:
+        print("Warning during creation:", operation.warnings, file=sys.stderr)
+    print(f"Instance {instance_name} created.")
+    return instance_client.get(project=project_id, zone=zone, instance=instance_name)
+
+
+# [END compute_preemptible_create]
+
+
+# [START compute_preemptible_check]
+def is_preemptible(project_id: str, zone: str, instance_name: str) -> bool:
+    """
+    Check if a given instance is preemptible or not.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone you want to use. For example: “us-west3-b”
+        instance_name: name of the virtual machine to check.
+    Returns:
+        The preemptible status of the instance.
+    """
+    instance_client = compute_v1.InstancesClient()
+    instance = instance_client.get(
+        project=project_id, zone=zone, instance=instance_name
+    )
+    return instance.scheduling.preemptible
+
+
+# [END compute_preemptible_check]
+
+
+# [START compute_preemptible_history]
+def preemption_history(project_id: str, zone: str, instance_name: str = None):
+    op_client = compute_v1.ZoneOperationsClient()
+    req = compute_v1.ListZoneOperationsRequest()
+    req.project = project_id
+    req.zone = zone
+
+    if instance_name:
+        req.filter = (
+            f'operationType="compute.instances.preempted" '
+            f"AND targetLink:instances/{instance_name}"
+        )
+    else:
+        req.filter = 'operationType="compute.instances.preempted"'
+
+    history = []
+
+    for operation in op_client.list(req):
+        this_instance_name = operation.target_link.rsplit("/", maxsplit=1)[1]
+        if instance_name and this_instance_name != instance_name:
+            # The filter used is not 100% accurate, it's `contains` not `equals`
+            # So we need to filter to make sure
+            continue
+        moment = operation.insert_time
+        history.append((instance_name, moment))
```

Comment:
Does this result in the same behavior as the code above? If so, I think it would be easier to follow than having the `continue`

```py
if instance_name and this_instance_name == instance_name:
    moment = operation.insert_time
    history.append((instance_name, moment))
```

Reply:
You are right, that's cleaner.